### PR TITLE
Add option to disable top site suggestions

### LINKED ISSUE #4977: Provide an option to disable "Top Sites" in autocomplete
## Test plan

https://github.com/brave/browser-laptop/pull/9063#issue-231507264

----

**Expected behavior:** Provide options to disable show Top Site and Brave Pages in "Search Bar Options" on `about:preferences#search`

screenshot 2016-10-20 13 33 39

- Platform (Win7, 8, 10? macOS? Linux distro?): OS X
- Brave Version: https://github.com/brave/browser-laptop/tree/8a174b537d75009585292ac4f7feb9354de3bbcc
- Any related issues:

### PR DESCRIPTION
**Pull request by luixxiul**
Closes #4977

Auditors:

Test Plan 1:
1. Open about:preferences#search
2. Disable `Show top site matches`
3. Open a new tab
4. Input `face`
5. Make sure that there does not appear the suggestions list

Test Plan 2:
1. Open about:preferences#search
2. Enable `Show top site matches`
3. Open a new tab
4. Input `face`
5. Make sure there appears the suggestions list, which has facebook.com

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by luixxiul** [comment]
Help wanted:

1. Clear the profile
2. Open `about:preferences#search`
3. Disable all of the options
4. Input `ja` on the URL bar

Actual result: `java.com` is autocompleted.

Expected result: nothing should be autocompleted as long as the top site matches are disabled.

File: app/common/lib/suggestion.js [hunk 1]
```diff
@@ -540,12 +540,15 @@ const getSearchSuggestions = (state, tabId, urlLocationLower) => {
 
 const getAlexaSuggestions = (state, urlLocationLower) => {
   return new Promise((resolve, reject) => {
+    const options = {
```
Comment (bbondy) [inline app/common/lib/suggestion.js]: Please follow the same pattern as here:
https://github.com/luixxiul/browser-laptop/blob/dd9ad5fe6077165d5d15df2428c5770a235f04d1/app/common/lib/suggestion.js#L504

Basically you resolve with an empty immutable list if the setting is not on.

**Comment by bbondy** [comment]
We're past the freeze for 0.16.100 so moving this to 0.16.200.  Only critical things and regressions are allowed past the freeze, thanks!

File: app/extensions/brave/locales/en-US/preferences.properties [hunk 2]
```diff
@@ -192,9 +192,10 @@ paintTabs=Show tabs in page theme color
 tabsPerTabPage=Number of tabs per tab set:
 tabCloseAction=When closing an active tab:
 showTabPreviews=Show tab previews on hover
+showOpenedTabMatches=Show tab matches
 showHistoryMatches=Show history matches
 showBookmarkMatches=Show bookmark matches
-showOpenedTabMatches=Show tab matches
+showTopsiteMatches=Show top site matches
```
Comment (cezaraugusto) [inline app/extensions/brave/locales/en-US/preferences.properties]: while it follows this prefs pattern, wouldn't be better to say "show top site suggestions" instead?

Comment (luixxiul) [inline app/extensions/brave/locales/en-US/preferences.properties]: It looks like reasonable. I'm addressing it.

Comment (luixxiul) [inline app/extensions/brave/locales/en-US/preferences.properties]: Addressed with 4844a1517

**Comment by luixxiul** [comment]
@cezaraugusto the two commits were squashed and rebased.

I'm going to replace the label with `ready-for-merge`.

**Comment by luixxiul** [comment]
thanks!

**Pull request merged**